### PR TITLE
feat(init-templates): add typesRoot compiler option for typescript templates in the tsconfig.json

### LINKED ISSUE #3830: Template tsconfig.json should specify typeRoots to avoid potential errors.
### General Information

* [X] :wave: I may be able to implement this feature request

* [ ] :warning: This feature might incur a breaking change

### Description

Given the following project structure:

- project_root/
   - deploy /
        - cdk generated
        - node_modules (cdk generated)
        - tsconfig.json
   - other_project_files
   - packages.json
   - node_modules

The deploy folder was empty, then the command `cdk init app --language=typescript` was executed.

The project_root contains the actual project.

Because of how `tsc` traverses the directories, if there is any issue in the `project_root/node_modules`  folder, compilation of the CDK (from within the deploy folder) will fail.

Currently we have a Gatsby project with a static website, for reasons beyond my understanding, a dependency is being pulled in and installed under `project_root\node_modules\@types\` that seems to have a compilation error. 

We don't use Typescript in the root project, but for some reason, when compiling the CDK App using `TSC` the compiler is making its way up to the parent folder.

### Proposed Solution

The only way I've been able to fix this was by specifying the typeRoots property in the CDK generated ` tsconfig.json` file by adding this under the "`compilerOptions`": 

```
"typeRoots": ["./node_modules/@types"]
```

This will ground the typescript compiler to the `deploy\node_modules` folder that contains all the CDK related things. Which is how it should probably work anyway.

### Environment

  - **CDK CLI Version: 1.6.0** 
  - **Module Version: 1.6.0** 
  - **OS: Windows 10 but most likely doesn't matter** 
  - **Language: Typescript**

### PR DESCRIPTION
**Pull request by carlosrfernandez**
Closes #3830 

Given a project structure where the cdk (init) generated `node_modules` has a parent node project with its own `node_modules` folder with failing typescript code, the CDK will not compile because `tsc` traverses the directory structure looking for type definitions.

This small change adds the `typesRoot` [compiler option](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#types-typeroots-and-types) essentially rooting the typescript compiler to look for types under the `cdk_root_folder/node_modules/@types`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/init-templates/sample-app/javascript/tsconfig.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "strictPropertyInitialization":false,
     "allowJs": true,
     "checkJs": true,
-    "noEmit": true
+    "noEmit": true,
+    "typeRoots": ["./node_modules/types"]
```
Comment (eladb) [inline packages/aws-cdk/lib/init-templates/sample-app/javascript/tsconfig.json]: shouldn't that be `@types`?

Comment (carlosrfernandez) [reply packages/aws-cdk/lib/init-templates/sample-app/javascript/tsconfig.json]: Jesus... yes it should be... you'd think I would have gotten it right on the first try.

**Comment by eladb** [comment]
@RomainMuller can you chime in? What do you think?

**Comment by RomainMuller** [comment]
That sounds fair - the only question is can this have unintended consequences?

**Comment by carlosrfernandez** [comment]
To be honest, I can’t really say if this will have additional side effects. 

But depending how you look at it, without having this setting, the side effects are already happening. 

It’s a valid concern though...

**Pull request merged**